src/objects/zcl_abapgit_objects.clas.abap

```diff
@@ -1346,7 +1346,9 @@ CLASS ZCL_ABAPGIT_OBJECTS IMPLEMENTATION.
     LOOP AT it_results ASSIGNING <ls_result> WHERE NOT obj_type IS INITIAL.
       IF <ls_result>-lstate IS NOT INITIAL
         AND NOT ( <ls_result>-lstate = zif_abapgit_definitions=>c_state-added
-        AND <ls_result>-rstate IS INITIAL ).
+        AND <ls_result>-rstate IS INITIAL )
+        OR ( <ls_result>-lstate IS INITIAL
+        AND <ls_result>-rstate = zif_abapgit_definitions=>c_state-deleted ).
```

Comment:
nice and symmetric, so guess its okay

but remote and local state doesn't work properly(and never will), I see the overall logic to be changed to something simple like, if pulling, pull the objects where there are diffs(ignoring the local and remote calculations), but its a different story

LGTM